### PR TITLE
TransportResyncReplicationAction should not honour blocks

### LINKED ISSUE #35597: Rolling-upgrade RecoveryIT tests are broken
The 6.x intake builds have been broken for most of today, with clusters failing to pass their wait conditions properly:

```
[ant:echo] ==> [Thu Nov 15 05:56:30 UTC 2018] checking health: http://[::1]:35001/_cluster/health?wait_for_nodes=>=1&wait_for_status=yellow
     [echo] ==> [Thu Nov 15 05:56:30 UTC 2018] checking health: http://[::1]:35001/_cluster/health?wait_for_nodes=>=1&wait_for_status=yellow
[ant:get] Getting: http://[::1]:35001/_cluster/health?wait_for_nodes=>=1&wait_for_status=yellow
      [get] Getting: http://[::1]:35001/_cluster/health?wait_for_nodes=>=1&wait_for_status=yellow
[ant:get] To: /var/lib/jenkins/workspace/elastic+elasticsearch+6.x+intake/qa/rolling-upgrade/build/cluster/v5.6.14#upgradedClusterTestCluster node0/cwd/wait.success
      [get] To: /var/lib/jenkins/workspace/elastic+elasticsearch+6.x+intake/qa/rolling-upgrade/build/cluster/v5.6.14#upgradedClusterTestCluster node0/cwd/wait.success

[ant:get] Error opening connection java.io.IOException: Server returned HTTP response code: 408 for URL: http://[::1]:35001/_cluster/health?wait_for_nodes=>=1&wait_for_status=yellow
      [get] Error opening connection java.io.IOException: Server returned HTTP response code: 408 for URL: http://[::1]:35001/_cluster/health?wait_for_nodes=>=1&wait_for_status=yellow
[ant:get] Error opening connection java.io.IOException: Server returned HTTP response code: 408 for URL: http://[::1]:35001/_cluster/health?wait_for_nodes=>=1&wait_for_status=yellow
```

This is from https://elasticsearch-ci.elastic.co/job/elastic+elasticsearch+6.x+intake/197/console although there are lots of other similar failures.

There's nothing obvious to me in the node logs themselves - the 3-node cluster looks to have formed fine. It does seem to reproduce locally, and @original-brownbear and I have been looking at this for a few hours without too much success. I will mute this test suite.

### PR DESCRIPTION
**Pull request by tlrx**
After #35332 has been merged, we noticed some test failures like #35597 in which one or more replica shards failed to be promoted as primaries because the primary replica re-synchronization never succeed.

After some digging it appeared that the execution of the resync action was blocked because of the presence of a global cluster block in the cluster state (in this case, the "no master" block), making the resync action to fail when executed on the primary. 

Until #35332 such failures never happened because the `TransportResyncReplicationAction` is skipping the reroute phase, the only place where blocks were checked. Now with #35332 blocks are checked during reroute and also during the execution of the transport replication action on the primary. After some internal discussion, we decided that the `TransportResyncReplicationAction` should never be blocked. This action is part of the replica to primary promotion and makes sure that replicas are in sync and should not be blocked when the cluster state has no master or when the index is read only.

This pull request changes the `TransportResyncReplicationAction` to make obvious that it does not honor blocks. It also adds a simple test that fails if the resync action is blocked during the primary action execution.

Closes #35597

### CODE REVIEW

File: server/src/main/java/org/elasticsearch/action/resync/TransportResyncReplicationAction.java [hunk 1]
```diff
@@ -102,6 +103,18 @@ protected void sendReplicaRequest(
         }
     }
 
+    @Override
+    protected ClusterBlockLevel globalBlockLevel() {
+        // resync should never be blocked
```
Comment (ywelsch) [inline server/src/main/java/org/elasticsearch/action/resync/TransportResyncReplicationAction.java]: maybe add here "because it's an internal action"

**Review by ywelsch — APPROVED** [review]
LGTM

File: server/src/main/java/org/elasticsearch/action/resync/TransportResyncReplicationAction.java (hunk 1 above)
Comment (tlrx) [reply server/src/main/java/org/elasticsearch/action/resync/TransportResyncReplicationAction.java]: Sure.

**Comment by tlrx** [comment]
Thanks @ywelsch

**Pull request merged**